src/core/thread_pool.rs

```diff
@@ -141,23 +141,7 @@ impl ThreadPool {
     }
 }
 
-// Safety: proof of the safety guarantees is deferred to the inner calls.
```

Comment:
Looks like you removed this safety comment, which Clippy now complains about. As in my previous review, please write `SAFETY` all in uppercase for consistency with the rest of the Paralight code base.